harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -123,6 +125,8 @@ def download(self, path: Optional[str] = None) -> str:
           - :func:`determined.estimator.load_estimator_from_checkpoint_path`
 
         Arguments:
+            via_master (bool): If true, checkpoints are downloaded through the master,
+                otherwise they are downloaded directly from checkpoint storage.
```

Comment:
Overall, I think it's a bad idea to make the user specify this value.  That's sort of leaking out implementation details into the user interface.

I would propose that `det checkpoint download UUID` first try and download UUID directly (which will be faster when it's available, and easier on us too), then fall back to trying to proxy it through the master, then give up.

I'm ok allowing a flag that lets the user pick proxy vs direct, but I think the default behavior needs to be smarter.

Reply:
I thought about that but there is a challenge: A user would either always have credentials or never. In the latter case, the user will always experience the overhead of trying to download directly and then fall back. And if the first attempt has any adverse effect that we fail to handle, the user will also experience it unnecessarily. It seems I can't sufficiently hide the implementation details and had better give users the choice. Suggestions?

Reply:
Ok.  I think that's a good case for _allowing_ the user to specify proxy vs direct, but not a good case to _require_ the user to do so.  It should be possible to handle failed direct downloads for like 99% of users so that most users never need to know about the `--via-master` flag.

Reply:
That sounds good. I can fall back to downloading via the master if direct downloads fail.